sdks/python/apache_beam/examples/complete/game/leader_board.py

```diff
@@ -0,0 +1,224 @@
+
+import apache_beam as beam
+from apache_beam.transforms import trigger
+
+
+def timestamp2str(t, fmt='%Y-%m-%d %H:%M:%S.000'):
+  """Converts a unix timestamp into a formatted string."""
+  return datetime.fromtimestamp(t).strftime(fmt)
+
+
+def parse_game_event(elem):
+  """Parses the raw game event info into a Python dictionary.
+
+  Each event line has the following format:
+    username,teamname,score,timestamp_in_ms,readable_time
+
+  e.g.:
+    user2_AsparagusPig,AsparagusPig,10,1445230923951,2015-11-02 09:09:28.224
+
+  The human-readable time string is not used here.
+  """
+  try:
+    row = list(csv.reader([elem]))[0]
+    yield {
+        'user': row[0],
+        'team': row[1],
+        'score': int(row[2]),
+        'timestamp': int(row[3]) / 1000.0,
+    }
+  except:  # pylint: disable=bare-except
+    logging.error('Parse error on "%s"', elem)
+
+
+class FormatTeamScores(beam.DoFn):
+  """Formats the data into the output format
+
+  Receives a (team, score) pair, extracts the window start timestamp, and
+  formats everything together into a string. Each string will be a line in the
+  output file.
+  """
+  def process(self, team_score, window=beam.DoFn.WindowParam):
+    team, score = team_score
+    start = timestamp2str(int(window.start))
+    yield 'window_start: %s, total_score: %s, team: %s' % (start, score, team)
+
+
+class CalculateTeamScoreSums(beam.PTransform):
+  """Calculates scores for each team within the configured window duration.
+
+  Extract team/score pairs from the event stream, using hour-long windows by
+  default.
+  """
+  def __init__(self, team_window_duration, allowed_lateness):
+    super(CalculateTeamScoreSums, self).__init__()
+    self.team_window_duration = team_window_duration * 60
+    self.allowed_lateness = allowed_lateness * 60
+
+  def expand(self, pcoll):
+    return (
+        # TODO: allowed_lateness not implemented yet in FixedWindows
+        # TODO: AfterProcessingTime not implemented yet, replace AfterCount
+        pcoll
+        | 'LeaderboardTeamFixedWindows' >> beam.WindowInto(
+            beam.window.FixedWindows(self.team_window_duration),
+            trigger=trigger.AfterWatermark(trigger.AfterCount(10),
+                                           trigger.AfterCount(20)),
+            accumulation_mode=trigger.AccumulationMode.ACCUMULATING)
+        | 'ExtractTeamScores' >> beam.Map(
+            lambda elem: (elem['team'], elem['score']))
+        | 'SumTeamScores' >> beam.CombinePerKey(sum)
+    )
+
+
+class CalculateUserScoreSums(beam.PTransform):
+  """Extract user/score pairs from the event stream using processing time, via
+  global windowing.
+
+  Get periodic updates on all users' running scores.
+  """
+  def __init__(self, allowed_lateness):
+    super(CalculateUserScoreSums, self).__init__()
+    self.allowed_lateness = allowed_lateness * 60
+
+  def expand(self, pcoll):
+    return (
+        # TODO: allowed_lateness not implemented yet in FixedWindows
```

Comment:
Add a note to the top that, the behavior does not exactly match Java and the website page.